src/Security/RolesMatrixBuilder.php

```diff
@@ -0,0 +1,335 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Security;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+class RolesMatrixBuilder
+{
+    /**
+     * @var TokenStorageInterface
+     */
+    protected $tokenStorage;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    protected $authorizationChecker;
+
+    /**
+     * @var Pool
+     */
+    protected $pool;
+
+    /**
+     * @var TranslatorInterface
+     */
+    protected $translator;
+
+    /**
+     * @var array
+     */
+    protected $rolesHierarchy;
+
+    /**
+     * @var array
+     */
+    private $labelPermission;
+
+    /**
+     * @var array
+     */
+    private $labelAdmin;
+
+    /**
+     * @var array
+     */
+    private $exclude;
+
+    /**
+     * @param TokenStorageInterface         $tokenStorage
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     * @param Pool                          $pool
+     * @param array                         $rolesHierarchy
+     */
```

Comment:
this comment can be removed, we have the same thing in constructor.

Reply:
👎 We should keep this for properties